README.md

```diff
@@ -1,73 +1,320 @@
+| subnets              | Defines a list of the subnets that should be available via the VPN connection.                    | list        | no       | -                       | leftsubnets/rightsubnets |
+
+#### public_key
+
+`public_key_src` specifies a path to a file on the controller host containing the public key used by this host for public key authentication without certificates. Otherwise, the user can directly specify the public key for this host by populating `public_key_content`. `public_key_content` can also accept a CKAID or nickname for a public key in the NSS database.
+
+Note that `public_key_src` and `public_key_content` may also be specified as host variables in the Ansible `all.hosts` section. The variable names in this case will be `vpn_public_key_src` and `vpn_public_key_content`. See [this example](#defining-public-keys-at-different-levels).
+
+**To be added later**: if neither `public_key_src` nor `public_key_content` is populated, the role will generate key pairs for each host.
+
+### Handling special cases with host names
+
+There are two ways to handle the special case when the names of the hosts in the Ansible inventory are not FQDN hostnames that you can use directly (e.g. they may be host aliases like "bastion_east"):  
+
+1. **Use the Ansible `hosts` configuration**
 
 ```yaml
-template_foo: "oof"
-template_bar: "baz"
+all:
+  hosts:
+    bastion_east:
+      ansible_host: bastion1.example.com
+      vpn_hostid: '@bastion-east'
+    bastion_west:
+      ansible_host: bastion2.example.com
+      vpn_hostid: '@bastion-west'
+    bastion_north:
+      ansible_host: bastion3.example.com
+      vpn_hostid: '@bastion-north'
+  vars:
+    vpn_connections:
+      - hosts:
+          bastion_east:
+          bastion_west:
+          bastion_north:
 ```
+The role would lookup the value of the host to use from `hostvars[hostname].ansible_host`.  You can specify other variables which are always associated with a particular host, like `vpn_hostid`, in the host configuration, which would be used for every connection where that host is referenced.
 
-### Variables Exported by the Role
+2. **Directly specify the host to use**
 
-This section is optional.  Some roles may export variables for playbooks to
-use later.  These are analogous to "return values" in Ansible modules.  For
-example, if a role performs some action that will require a system reboot, but
-the user wants to defer the reboot, the role might set a variable like
-`template_reboot_needed: true` that the playbook can use to reboot at a more
-convenient time.
+Users can provide the host specific parameters with either the host's definition in the Ansible `all.hosts` section, or under each host in the `hosts` list under `vpn_connections`.  This gives the admin flexibility in case they cannot edit values in one or the other section.
```

Comment:
It does not seem correct to speak about `all.hosts` section. Shouldn't one be able to provide the host specific parameter as any host-scoped variable? There are may ways to set host-scoped variables, and even if they are set in the inventory, the host does not need to be listed under the `all` group, and the variable does not need to be set on a host (it can be a group variable). The inventory does not even need to be in the YAML format (it can be in the INI format).